lumeer-core/src/main/java/io/lumeer/core/cache/UserCache.java

```diff
@@ -38,25 +38,29 @@
    @Inject
    private UserDao userDao;
 
+   @Inject
+   private WorkspaceKeeper workspaceKeeper;
+
    private Cache<User> userCache;
 
    @PostConstruct
    public void initCache() {
       userCache = cacheFactory.getCache();
    }
 
-   public User getUser(String username) {
-      return userCache.computeIfAbsent(username, this::getOrCreateUser);
+   public User getUser(String email) {
+      return userCache.computeIfAbsent(email, this::getOrCreateUser);
    }
 
-   private User getOrCreateUser(String username) {
-      Optional<User> userOptional = userDao.getUserByUsername(username);
+   private User getOrCreateUser(String email) {
+      String organizationId = workspaceKeeper.getOrganization().get().getId(); // TODO how to be sure, that organization is set?
```

Comment:
Once you fix the problem with DAO methods you will see there is no need for getting the organization here.